src/node_crypto.cc

```diff
@@ -4825,6 +4826,429 @@ void Scrypt(const FunctionCallbackInfo<Value>& args) {
 #endif  // OPENSSL_NO_SCRYPT
 
 
+class KeyPairGenerationConfig {
+ public:
+  virtual EVP_PKEY_CTX* Setup() = 0;
+  virtual bool Configure(const EVPKeyCtxPointer &ctx) {
+    return true;
+  }
+};
+
+class RSAKeyPairGenerationConfig : public KeyPairGenerationConfig {
+ public:
+  RSAKeyPairGenerationConfig(unsigned int modulus_bits, unsigned int exponent)
+    : modulus_bits_(modulus_bits), exponent_(exponent) {}
+
+  EVP_PKEY_CTX* Setup() {
+    return EVP_PKEY_CTX_new_id(EVP_PKEY_RSA, nullptr);
+  }
+
+  bool Configure(const EVPKeyCtxPointer &ctx) {
+    if (EVP_PKEY_CTX_set_rsa_keygen_bits(ctx.get(), modulus_bits_) <= 0)
+      return false;
+
+    if (exponent_ != 0x10001) {
```

Comment:
If I understand correctly, this relies on the fact that the default exponent is 65537. Could that change? What happens if it does?

Reply:
I added a comment. If OpenSSL ever decides to switch away from `0x10001`, we can simply remove the check and always specify the exponent.